tests/serde.rs

```diff
@@ -214,10 +230,24 @@ fn enums() {
     test_enum! {
         ExternallyTagged
     }
-    test_enum! {
-        #[serde(tag = "tag")]
-        InternallyTagged
+
+    #[derive(Debug, PartialEq, Serialize, Deserialize)]
+    #[serde(tag = "tag")]
+    enum InternallyTagged<A, B> {
+        Unit,
+        Struct { a: A, b: B },
     }
+
+    test_via_json(InternallyTagged::Unit::<(), ()>);
```

Comment:
It can't use the `test_enum!` macro as it is for internally tagged enums since serde doesn't support internal tags on tuple variants. 

There was also a problem with the NewType variant since internally tagged enums work with a struct as the new type, but not something like a string (which is what the `test_enum!` macro was testing it with). This is because serde has to insert the tag inside the key/value pairs of the struct in a new-type variant.

I'm open to testing this enum in a different way though.

Reply:
Ah yeah, this makes sense. Thanks for clarification.